### PR TITLE
Bump version of Dart >=3.0.2 and Flutter >=3.10.2

### LINKED ISSUE #1307: Bump version of Dart to 3.0.2 and Flutter to 3.10.2
Bump the versions of:
Dart ^3.0.2
Flutter ^3.10.2

### PR DESCRIPTION
**Pull request by desistefanova**
Bump the versions of:

- Dart ^3.0.2
- Flutter ^3.10.2
- ffigen: ^8.0.2

Fixes #1307

### CODE REVIEW

File: lib/src/native/decimal128.dart [hunk 1]
```diff
@@ -19,7 +19,7 @@
 part of 'realm_core.dart';
 
 /// A 128-bit decimal floating point number.
-class Decimal128 extends Comparable<Decimal128> implements common.Decimal128 {
+class Decimal128 implements Comparable<Decimal128>, common.Decimal128 {
```
Comment (desistefanova) [inline lib/src/native/decimal128.dart:22]: @nielsenko we need your agreement for this change. Could it cause some issues?

**Review by nielsenko — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@
 
 ### Compatibility
 * Realm Studio: 13.0.0 or later.
+* Dart >=3.0.2 <4.0.0 (Flutter >=3.10.2) on Android, iOS, Linux, MacOS and Windows
```
[the remark this answers is not in the document]
Comment (blagoev) [reply CHANGELOG.md:11]:
```suggestion
* Dart >=3.0.2 <4.0.0 (Flutter >=3.10.2)
```

**Review by blagoev — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,7 @@
 
 ### Compatibility
 * Realm Studio: 13.0.0 or later.
+* Dart >=3.0.2 <4.0.0 (Flutter >=3.10.2)
```
Comment (desistefanova) [inline CHANGELOG.md:11]:
```suggestion
* Dart ^3.0.2  and Flutter ^3.10.2
```

Comment (desistefanova) [reply CHANGELOG.md:11]: @blagoev how about that?` * Dart ^3.0.2 and Flutter ^3.10.2`

**Pull request merged**